qemu/tests/numa_cpu.py

```diff
@@ -0,0 +1,142 @@
+        for node in nodes_guest:
+            numa_cpus = numa_info_guest.online_nodes_cpus[node]
+            numa_cpus = set([int(v) for v in numa_cpus.split()])
+            numa_cpu_guest.append(numa_cpus)
+        return numa_cpu_guest
+
+    def numa_cpu_cli():
+        """
+        Get the numa cpus for each node according to the qemu cli
+        """
+        error_context.context("Get the expected cpus in qemu command line", logging.info)
+        cpus = params.objects("guest_numa_cpus")
+        numa_cpu_cli = []
+        tmp = {}
+        for cpu in cpus:
+            cpu_params = params.object_params(cpu)
+            nodeid = cpu_params["numa_cpu_nodeid"]
+            socket = cpu_params.get("numa_cpu_socketid")
+            die = cpu_params.get("numa_cpu_dieid")
+            core = cpu_params.get("numa_cpu_coreid")
+            thread = cpu_params.get("numa_cpu_threadid")
+            cpu_list = convert_cpu_topology_to_index(socket, die, core, thread)
+            if nodeid in tmp.keys():
+                tmp[nodeid] += cpu_list
+            else:
+                tmp[nodeid] = cpu_list
+        for item in sorted(tmp.items(), key=lambda item: item[0]):
+            numa_cpu_cli.append(set(sorted(item[1])))
+        return numa_cpu_cli
+
+    def get_hotpluggable_cpus():
+        """
+        Parse output of "query-hotpluggable-cpus", get the numa cpus for each node.
+        """
+        error_context.context("Get the hotpluggable cpus", logging.info)
+        binded_cpus = []
+        tmp = {}
+        out = vm.monitor.info("hotpluggable-cpus")
+        for vcpu_info in out:
+            vcpus_count = vcpu_info["vcpus-count"]
+            vcpu_info = vcpu_info["props"]
+            nodeid = vcpu_info.get("node-id")
+            socketid = vcpu_info.get("socket-id")
+            dieid = vcpu_info.get("die-id")
+            coreid = vcpu_info.get("core-id")
+            threadid = vcpu_info.get("thread-id")
+            if nodeid is not None:
+                if vm_arch in ('x86_64', 'i386'):
+                    cpu_index = [socketid * socket_weight + dieid * die_weight +
+                                 coreid * core_weight + threadid * thread_weight]
+                elif vm_arch in ('ppc64', 'ppc64le'):
+                    cpu_index = range(coreid, coreid + vcpus_count)
+                if nodeid in tmp.keys():
+                    tmp[nodeid] += cpu_index
+                else:
+                    tmp[nodeid] = cpu_index
+        for item in sorted(tmp.items(), key=lambda item: item[0]):
+            binded_cpus.append(set(sorted(item[1])))
+        return binded_cpus
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    os_type = params["os_type"]
+    vm_arch = params["vm_arch_name"]
+
+    vcpu_threads = params.get_numeric('vcpu_threads')
+    if vm_arch in ('x86_64', 'i386'):
+        vcpu_sockets = params.get_numeric('vcpu_sockets')
+        vcpu_dies = params.get_numeric('vcpu_dies')
```

Comment:
For earlier qemu, there might not be dies option.

Reply:
> For earlier qemu, there might not be dies option.

Already use `required_qemu = [4.1.0, )` with comments. Do you have any concern with that ?